File: FizzBuzz/Python_2/FizzBuzzTest.py
import unittest
import FizzBuzz


class FizzBuzzTest(unittest.TestCase):
    def setUp(self):
        self.sut = FizzBuzz.FizzBuzz()

    def test_Say_NumberNotDivisible3Or5_ReturnNumber(self):
        # arrange
        expected = '1'

        # act
        result = self.sut.Say(1)

        # assert
        self.assertEqual(expected, result)

    def test_Say_NumberIsDivisible3_ReturnFizz(self):
        # arrange
        expected = 'Fizz'

        # act
        result = self.sut.Say(3)

        # assert
        self.assertEqual(expected, result)

    def test_Say_NumberIsDivisible5_ReturnBuzz(self):
        # arrange
        expected = 'Buzz'

        # act
        result = self.sut.Say(5)

        # assert
        self.assertEqual(expected, result)

    def test_Say_NumberIsDivisible3And5_ReturnFizzBuzz(self):
        # arrange
        expected = 'FizzBuzz'

        # act
        result = self.sut.Say(15)

        # assert
        self.assertEqual(expected, result)


if __name__ == '__main__':
    unittest.main

File: FizzBuzz/C#_5/FizzBuzzTest.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using NUnit.Framework;

namespace Kata5
{
    [TestFixture]
    internal class FizzBuzzTest
    {
        private FizzBuzz _sut;

        [OneTimeSetUp]
        public void Init()
        {
            this._sut = new FizzBuzz();
        }

        [Test]
        [TestCase(1)]
        public void Say_NumberNotDivisible3Or5_ReturnNumber(int number)
        {
            // Arrange
            // Act
            var result = this._sut.Say(number);

            // Assert
            Assert.AreEqual(number.ToString(), result);
        }

        [Test]
        [TestCase(3)]
        public void Say_NumberIsDivisible3_ReturnFizz(int number)
        {
            // Arrange
            // Act
            var result = this._sut.Say(number);

            // Assert
            Assert.AreEqual("Fizz", result);
        }

        [Test]
        [TestCase(5)]
        public void Say_NumberIsDivisible5_ReturnBuzz(int number)
        {
            // Arrange
            // Act
            var result = this._sut.Say(number);

            // Assert
            Assert.AreEqual("Buzz", result);
        }

        [Test]
        [TestCase(15)]
        public void Say_NumberIsDivisible3And5_ReturnFizzBuzz(int number)
        {
            // Arrange
            // Act
            var result = this._sut.Say(number);

            // Assert
            Assert.AreEqual("FizzBuzz", result);
        }

    }
}
